thrust/testing/merge_by_key.cu

```diff
@@ -120,47 +102,68 @@ void TestMergeByKeyDispatchImplicit()
 
   ASSERT_EQUAL(13, vec.front());
 }
+
+template <typename T, typename CompareOp, typename... Args>
+auto call_merge_by_key(Args&&... args) -> decltype(thrust::merge_by_key(std::forward<Args>(args)...))
+{
+  _CCCL_IF_CONSTEXPR (::cuda::std::is_void<CompareOp>::value)
+  {
+    return thrust::merge_by_key(std::forward<Args>(args)...);
+  }
+  else
+  {
+    // TODO(bgruber): remove next line in C++17 and pass CompareOp{} directly to stable_sort
+    using C = ::cuda::std::__conditional_t<::cuda::std::is_void<CompareOp>::value, thrust::less<T>, CompareOp>;
+    return thrust::merge_by_key(std::forward<Args>(args)..., C{});
+  }
+  _LIBCUDACXX_UNREACHABLE();
+}
+
 DECLARE_UNITTEST(TestMergeByKeyDispatchImplicit);
 
-template <typename T>
+template <typename T, typename CompareOp = void>
 void TestMergeByKey(size_t n)
 {
-  thrust::host_vector<T> random_keys = unittest::random_integers<unittest::int8_t>(n);
-  thrust::host_vector<T> random_vals = unittest::random_integers<unittest::int8_t>(n);
-
-  size_t denominators[]   = {1, 2, 3, 4, 5, 6, 7, 8, 9};
-  size_t num_denominators = sizeof(denominators) / sizeof(size_t);
+  const auto random_keys = unittest::random_integers<unittest::int8_t>(n);
+  const auto random_vals = unittest::random_integers<unittest::int8_t>(n);
 
-  for (size_t i = 0; i < num_denominators; ++i)
+  const size_t denominators[] = {1, 2, 3, 4, 5, 6, 7, 8, 9};
+  for (const auto& denom : denominators)
   {
-    size_t size_a = n / denominators[i];
+    const size_t size_a = n / denom;
 
     thrust::host_vector<T> h_a_keys(random_keys.begin(), random_keys.begin() + size_a);
     thrust::host_vector<T> h_b_keys(random_keys.begin() + size_a, random_keys.end());
 
-    thrust::host_vector<T> h_a_vals(random_vals.begin(), random_vals.begin() + size_a);
-    thrust::host_vector<T> h_b_vals(random_vals.begin() + size_a, random_vals.end());
+    const thrust::host_vector<T> h_a_vals(random_vals.begin(), random_vals.begin() + size_a);
+    const thrust::host_vector<T> h_b_vals(random_vals.begin() + size_a, random_vals.end());
```

Comment:
I am wondering, we are always just sorting subranges of the same vector. Could we just use a `cuda::std::span` and avoid the separate vectors?

Just kidding we still have C++11